src/main/java/dk/cngroup/lentils/service/ImageService.java

```diff
@@ -157,48 +158,73 @@ public void deleteImageFile(final Image image) {
     }
 
     public String getImageUrlForHintDto(final Hint hint) {
-        return hint.getImage() == null || hint.getImage().isFromFile() ? "" : hint.getImage().getImageUrl();
+        return hint.getImage() == null || hint.getImage().isLocal() ? "" : hint.getImage().getImageUrl();
     }
 
     public ImageSource getImageSource(final Hint hint) {
        if (!Optional.ofNullable(hint.getImage()).isPresent()) {
            return ImageSource.NONE;
        }
-       if (hint.getImage().isFromFile()) {
-           return ImageSource.FILE;
+       if (hint.getImage().isLocal()) {
+           return ImageSource.LOCAL;
        }
-       return ImageSource.WEB;
+       return ImageSource.REMOTE;
     }
 
     public Image generateImageFromDto(final String imageUrl,
                                       final ImageSource imageSource,
                                       final MultipartFile file,
                                       final Image actualImage) {
-        Image image = null;
         switch (imageSource) {
-            case WEB:
-                if (actualImage == null) {
-                    return new Image(imageUrl, false);
-                }
-                deleteImageFile(actualImage);
-                actualImage.setImageUrl(imageUrl);
-                actualImage.setFromFile(false);
-                return actualImage;
-            case FILE:
-                if (file.isEmpty()) {
-                    return actualImage;
-                }
-                if (actualImage == null) {
-                    image = createImageFromMultipartFile(file);
-                    saveFile(image.getImageUrl(), file);
-                    return image;
-                }
-                deleteImageFile(actualImage);
-                setImageFromMultipartFile(file, actualImage);
-                saveFile(actualImage.getImageUrl(), file);
-                return actualImage;
+            case REMOTE:
+                return getImageForRemoteSource(imageUrl, actualImage);
+            case LOCAL:
+                return getImageFromLocalSource(file, actualImage);
             default:
-                return image;
+                return null;
         }
     }
+
+    @NotNull
+    private Image getImageFromLocalSource(final MultipartFile file, final Image actualImage) {
```

Comment:
Tady se mi nelibi, ze existuji metody, ktere delaji to same, jen jedno pro remote file a druha pro local. Toto samo o sobe uz znaci, ze by se mel pouzit polymorfismus. Mohly by existovat dve sluzby LocalImageService a RemoteImageService, ktere obe budou implementovat metody `getImage`, `setNew` a mozna dalsi. Podle typu vy se vybrala patricna sluzba a uz by se kod nedelil. Pripadne se to da resit i pres Localmage a RemoteImage a extrakci logiky do techto trid. Zalezi, jak se ti to bude vic pozdavat.

Reply:
predelala jsem to, ale nejsem si jista, jestli je to lepsi. Mam tam ImageManipulation (blby nazev, ale na nic jineho jsem neprisla) a 2 implementace pro local a remote, ty musim vytvaret rucne. Dost jsem toho kvuli tomu musela predelat, nakonec je tam jen 1 metoda